changed the usaco guide code editing rules (#1684)
cpinitiative/usaco-guide: content/1_General/Adding_Solution.mdx

    ```

    Keep file names and solution IDs consistent; namely, USACO IDs should
    correspond to the last digits at the very end of the URL on

>     [usaco.org](http://www.usaco.org/index.php?page=viewproblem2&cpid=690).

Reviewer: ```suggestion
    [usaco.org](http://www.usaco.org/index.php?page=viewproblem2&cpid=690). See [Working With MDX](/general/working-mdx#problem-lists) for more examples of IDs.
```
Is this the latest version of the file? the version on the guide doesn't exactly match what you have here. can you merge master into this branch?

3.  Add your implementation, following these conventions and the conventions
    [below](#code-conventions):

>     - Use _tabs_ to indent (If your code is written with spaces, click on the usaco guide editor [https://usaco.guide/editor/](https://usaco.guide/editor/), copy-	paste your code in, right click, click "command pallete" from the menu that pops up and then search for "convert indentation to tabs". Once you see it, click       on it and then copy that new code with the correct indentation onto your github pull request)

Reviewer: ```suggestion
    - Use _tabs_ to indent
    <Info title="Pro Tip">
    If your code is written with spaces, use the [Guide Editor](https://usaco.guide/editor/), copy-paste your code in, right click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation into your Github pull request.
    </Info>
```

>     - Lines should not exceed 80 characters. If you need to, you can write comments accros mutliple lines to follow this rule.

Reviewer 2: ```suggestion
    - Lines should not exceed 80 characters. If you need to, you can write comments across multiple lines to follow this rule.
```

    - Your code needs to have comments if it doesn't contribute anything to an already existing solution on the guide or official USACO solution. Comments

>       shouldn't be used for every single obvious line and should only describe complicated or big chunks of code. Comments should be short and consise, as in being       not longer than 2-3 sentences but still encompassing all the valuable information an inexperienced coder might not be able to automatically assume. 

Reviewer 2: ```suggestion
      shouldn't be used for every single obvious line and should only describe complicated or big chunks of code. Comments should be short and concise, as in being not longer than 2-3 sentences but still encompassing all the valuable information an inexperienced coder might not be able to automatically assume. 
```

    - Be _consistent_ about formatting your code, unlike the following snippet.

>       We don't care about which one you use (`a+b` or `a + b`), but pick one and stick with it throught your entire code!

Reviewer: ```suggestion
      We don't care about which one you use (`a+b` or `a + b`), but pick one and stick with it throughout your entire code!
```

     ```py
      a,b = (0,1)
      print(a+b)
      print(a + b)
      ```

>     - Use universally understandable variable and data structure names! Simply calling an integer that serves more purpose than just the input or output as "x" or       "sus" or "ThisismyvariablethatonlyIcanunderstandgoodbyeandgoodday" will not suffice. Please use variable names that also make sense in respect to the

Reviewer 2: ```suggestion
    - Use universally understandable variable and data structure names! Simply calling an integer that serves more purpose than just the input or output as "x" or "sus" or "ThisismyvariablethatonlyIcanunderstandgoodbyeandgoodday" will not suffice. Please use variable names that also make sense in respect to the
```

Reviewer 2: can we be more serious here

Author: > can we be more serious here

done

      problem that you are writing your solution to. 

>     - Remeber, Your Pull Request will not be accepted until every single aspect of your code is close to perfect, so please try to follow these rules to get it           right the first time.

Reviewer 2: ```suggestion
    - Remember, your Pull Request will not be accepted until every single aspect of your code is close to perfect, so please try to follow these rules to get it right the first time.
```

    ```

    Keep file names and solution IDs consistent; namely, USACO IDs should
    correspond to the last digits at the very end of the URL on
    [usaco.org](http://www.usaco.org/index.php?page=viewproblem2&cpid=690) and

>     file names should be the same as the solution's ID.

Reviewer: ```suggestion
    file names should be the same as the solution's ID. See
    [Working With MDX](/general/working-mdx#problem-lists) for more examples of
    IDs.
```

    ```

    Keep file names and solution IDs consistent; namely, USACO IDs should

>     correspond to the last digits at the very end of the URL on

Reviewer: ```suggestion
    correspond to the number at the very end of the URL on
```

    [usaco.org](http://www.usaco.org/index.php?page=viewproblem2&cpid=690) and
    file names should be the same as the solution's ID. See
    [Working With MDX](/general/working-mdx#problem-lists) for more examples of
    IDs.

3.  Add your implementation, following these conventions and the conventions
    [below](#code-conventions):

    - Use _tabs_ to indent
    <Info title="Pro Tip">

> 	If your code is written with spaces, use the [Guide Editor](/editor), copy-paste your code in, right-click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation into your Github pull request.

Reviewer: ```suggestion
	If your code is indented with spaces, use the [Guide Editor](/editor), copy-paste your code in, right-click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation into your Github pull request.
```

> 	If your code is written with spaces, use the [Guide Editor](https://usaco.guide/editor/), copy-paste your code in, right click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation into your Github pull request.

Reviewer 2: ```suggestion
	If your code is written with spaces, use the [Guide Editor](/editor), copy-paste your code in, right-click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation into your Github pull request.
```

> 	If your code is written with spaces, use the [Guide Editor](https://usaco.guide/editor/), copy-paste your code in, right click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation. into your Github pull request.

Reviewer 3: ```suggestion
	If your code is written with spaces, use the [Guide Editor](https://usaco.guide/editor/), copy-paste your code in, right click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation into your Github pull request.
```

    </Info>
    
    - Lines should not exceed 80 characters. If you need to, you can write comments across multiple lines to follow this rule.

>     - Your code needs to have comments if it doesn't contribute anything to an already existing solution on the guide or official USACO solution. Comments shouldn't be used for every single obvious line and should only describe complicated or big chunks of code. Comments should be short and concise, being not longer than 2-3 sentences but still encompassing all the valuable information an inexperienced coder might not be able to automatically assume. 

Reviewer 3: ```suggestion
    - Add comments throughout your code to explain your logic. Feel free to copy-paste the official USACO solution and improve it by adding more descriptive variable names and helpful comments.
```

oops idk what to say about comments; I think it's difficult to create hard rules regarding commenting. I'm open to suggestions

Reviewer 3: > having a list of every trait a good comment encompasses would be helpful.

idk if this is possible...

> Your code needs to have comments

only when it makes sense to have comments, right?

> Comments shouldn't be used for every single obvious line ... Comments should be short and concise

this (should) be obvious...

> being not longer than 2-3 sentences

this is not necessarily always true?

Author: you are right hold a vote or something to tell me what to write for the comments guidelines. 
if a) i should include most of what I said reformatted in bullet points so its easier to read or b) commit your changes or c) asking anyone what their idea should be

Reviewer 2: ye ur code doesnt always need comments

comments can be longer than 3 sentences tho rare

comments doesnt have a set rule to follow it just explains code

Reviewer 2: personally i think Nathans changes should be committed

Reviewer 4: one thing i suggest we should add is that comments shouldn't explain what your code does, but instead explain why it does what it does.

Author: commited

    - Be _consistent_ about formatting your code, unlike the following snippet.
      We don't care about which one you use (`a+b` or `a + b`), but pick one and stick with it throughout your entire code!

     ```py
      a,b = (0,1)
      print(a+b)
      print(a + b)
      ```

>     - Use universally understandable variable and data structure names! This is especially true for lower divisions (ie. Bronze and Silver). It is preferred to use longer / more descriptive variable names than what you would normally use in-contest, since this is a learning resource.

Reviewer 2: ```suggestion
    - Use universally understandable variable names, especially for lower divisions (ie. Bronze and Silver). It is preferred to use longer / more descriptive variable names than what you would normally use in-contest since this is a learning resource.
```

>     - Remember to follow these guidelines when formatting your code for a pull request. These rules all have a specific purpose that makes the code readable to a general audience. Making sure that your pull request encompasses all these details would be desirable.

Reviewer 3: wait what is even the point of this last bullet point? it should be very self explanatory that one should follow guidelines right? 😆 

other than that looks good!

Author: removing it rn

Author: done

>     - Use universally understandable variable and data structure names! Simply calling an integer that serves more purpose than just the input or output as "x" or "sus" will not suffice. Please use variable names that also make sense in respect to the problem that you are writing your solution to. 

Reviewer 3: ```suggestion
    - Use universally understandable variable and data structure names! This is especially true for lower divisions (ie. Bronze and Silver). It is preferred to use longer / more descriptive variable names than what you would normally use in-contest, since this is a learning resource.
```

>     - Remember to follow these guidelines when formatting your code for a pull request. These rules all have a specific purpose that make the code readable to a general audience. Making sure that your code encompasses all these details on the first submission would be desirable, but you may make as many commits as you need once feedback is given by team/core members. 

Reviewer 2: ```suggestion
    - Remember to follow these guidelines when formatting your code for a pull request. These rules all have a specific purpose that makes the code readable to a general audience. Making sure that your pull request encompasses all these details would be desirable.
```

>     - Remember, your pull request will not be accepted until every single aspect of your code is close to perfect, so please try to follow these rules to get it right the first time.

Reviewer 3: I don't really know if this line is necessary? idk :shrug:

Author: it is because i didn't think that you guys would adhere to these guidelines so strictly so I went kinda loose with them and ended up having to redo most of my code multiple times because of it.

Reviewer 2: can u change like the until it is almost perfect part. doesnt sound very nice xD

Reviewer 3: idk I just don't see the point and I feel like this line gives off bad vibes 😛

Reviewer 2: yea that's what i'm feeling too xd

Author: okay ill make it sound nicer and also more warm
